contrib/segmentio/kafka-go/headers.go

```diff
@@ -0,0 +1,48 @@
+package kafka
+
+import (
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/segmentio/kafka-go"
+)
+
+// A MessageCarrier injects and extracts traces from a sarama.ProducerMessage.
+type MessageCarrier struct {
```

Comment:
Do we need a struct? Probably not
```suggestion
// contextCarrier implements tracer.TextMapWriter and tracer.TextMapReader on top of a kafka.Message.
type contextCarrier *kafka.Message
```

Reply:
Yes, I verified it, it doesn't update the header when we use `contextCarrier`